src/socket.c

```diff
@@ -731,26 +719,43 @@ void pthreads_socket_select(zval *read, zval *write, zval *except, uint32_t sec,
 	}
 
 	PHP_SAFE_MAX_FD(mfd, 0);
-	
-	if (sec || usec) {
+
+	/* If seconds is not set to null, build the timeval, else we wait indefinitely */
+	if (sec != NULL) {
```

Comment:
From what I can tell, `sec` can never be `NULL`, since it is not an optional parameter (the same check in php-src looks superfluous).

Reply:
`sec` needs to be `null`able. The following applies to `Socket::select()` and `socket_select()`

- `sec` < 0 - Invalid argument
- `sec` = 0 - returns immediately
- `sec` > 0 - waits definitely
- `sec` = null - waits indefinitely

Reply:
Right, I see. I didn't realise `sec` would be `NULL` (C null) if a PHP `null` was passed in to it (I assumed it would have been `Z_TYPE_P(sec) == IS_NULL`). My mistake...